graphql/util.go

```diff
@@ -1163,3 +1165,65 @@ func interfaceToMap(ctx context.Context, data interface{}) (map[string]interface
 
 	return mapField, nil
 }
+
+type HasMatchingTasks struct {
+	Id       string
+	HasTasks bool
+}
+
+func concurrentlyBuildHasMatchingTasksMap(ctx context.Context, versions []model.Version, opts task.HasMatchingTasksOptions, hasMatchingTasksMap map[string]bool) error {
```

Comment:
I would advocate for returning a `map[string]bool` instead of passing one. This way you can return `nil` in the case where you hit any errors. I don't have a particularly strong argument though other than that typically in Go you would expect the results of a function in the return values.